### ISSUE TITLE
NPE when preview large amount of data

### ISSUE THREAD

**Issue by instcode** [embulk/embulk#570]
When preview data from command line using: `embulk preview -Ilib load-shared_products.yml`, the NPE occurs:
```
2017-03-24 11:20:58.212 +0700: Embulk v0.8.16
2017-03-24 11:20:58.966 +0700 [INFO] (0001:preview): Loaded plugin embulk/input/app_annie from a load path
2017-03-24 11:20:59.028 +0700 [INFO] (0001:preview): Logging initialized @2342ms
java.lang.NullPointerException
	at org.embulk.spi.util.Pages.toObjects(org/embulk/spi/util/Pages.java:24)
	at org.embulk.command.PreviewPrinter.printAllPages(org/embulk/command/PreviewPrinter.java:36)
	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:497)
	at org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(org/jruby/javasupport/JavaMethod.java:453)
	at org.jruby.javasupport.JavaMethod.invokeDirect(org/jruby/javasupport/JavaMethod.java:314)
	at RUBY.preview(~/.embulk/bin/embulk!/embulk/runner.rb:48)
	at RUBY.run(~/.embulk/bin/embulk!/embulk/command/embulk_run.rb:305)
	at RUBY.<main>(~/.embulk/bin/embulk!/embulk/command/embulk_main.rb:2)
	at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:850)
	at org.jruby.Ruby.loadFile(org/jruby/Ruby.java:2976)
	at org.jruby.RubyKernel.requireCommon(org/jruby/RubyKernel.java:963)
	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:956)
	at org.jruby.RubyKernel$INVOKER$s$1$0$require19.call(org/jruby/RubyKernel$INVOKER$s$1$0$require19.gen)
	at RUBY.(root)(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1)
	at ~.$_dot_embulk.bin.embulk.embulk.command.embulk_bundle.invokeOther66:require(~/$_dot_embulk/bin/embulk/embulk/command/file:~/.embulk/bin/embulk!/embulk/command/embulk_bundle.rb:51)
	at ~.$_dot_embulk.bin.embulk.embulk.command.embulk_bundle.<main>(file:~/.embulk/bin/embulk!/embulk/command/embulk_bundle.rb:51)
	at java.lang.invoke.MethodHandle.invokeWithArguments(java/lang/invoke/MethodHandle.java:625)
	at org.jruby.Ruby.runScript(org/jruby/Ruby.java:834)
	at org.jruby.Ruby.runNormally(org/jruby/Ruby.java:749)
	at org.jruby.Ruby.runNormally(org/jruby/Ruby.java:767)
	at org.jruby.Ruby.runFromMain(org/jruby/Ruby.java:580)
	at org.jruby.Main.doRunFromMain(org/jruby/Main.java:425)
	at org.jruby.Main.internalRun(org/jruby/Main.java:313)
	at org.jruby.Main.run(org/jruby/Main.java:242)
	at org.jruby.Main.main(org/jruby/Main.java:204)
	at org.embulk.cli.Main.main(org/embulk/cli/Main.java:23)

Error: 
```
(no more text after `Error:` above)

In order to product the issue, please use `embulk-input-app_annie-0.1.5.gem` with the following load-shared_products.yml:
```
in:
  type: app_annie
  apikey: xxxxxxxxxxxxxxxxxxxxxxxx 
  target: shared_products
out: {type: stdout}
```

It seems that the error occurs if the amount of data is big (100+ records). If I limit the number of output records to 15, the error is gone.

**Comment by hiroyuki-sato** [comment]
@instcode 
Can you execute `run` command normally?.
If so, you may hit #476 issue. 
And I PRed #564 for the same issue.
Please check 32KB limitation.

**Comment by instcode** [comment]
@hiroyuki-sato Yes, I can execute `run` normally. So it must be an issue with preview function. 32KB limitation in #476 would be the root cause but I'm not sure though as the exception here is NPE.

**Comment by hiroyuki-sato** [comment]
@instcode Thank you for the investigating this issue.

If you provide us reproduce code, please let me know. 
I couldn't find `embulk-input-app_annie` code. 
We need to fix case #476. Additionally, may require more investigation about this issue.

Finally, Please vote 👍 #476 for escalate this issue.

**Comment by hiroyuki-sato** [comment]
The `embulk-input-app_annie` is the Treasure Data internal project.
That's why I can't access that project.
